schema.prisma

```diff
@@ -0,0 +1,682 @@
+  id    String @id @db.VarChar(1)
+  /// 下拉選單順序，來自舊系統 `Cl_seq`
+  order Int
+
+  courses      Course[]
+  departments  Department[]
+  translations GradeTranslation[]
+}
+
+model GradeTranslation {
+  language Language
+
+  /// 年級名稱
+  name String
+
+  gradeId String @db.VarChar(1)
+  grade   Grade  @relation(fields: [gradeId], references: [id])
+
+  @@id([language, gradeId])
+}
+
+/// 來自舊系統 NOL table `ifcrfcr`
+model Classroom {
+  id            String  @id @db.Uuid
+  /// 教室編號
+  name          String  @db.VarChar(9)
+  location      String  @db.VarChar(22)
+  buildingName  String? @db.VarChar(32)
+  /// 用於產生外部地圖連結
+  buildingId    String  @db.VarChar(8)
+  accessibility Boolean // TODO: bool or enum?
+
+  // TODO: 是否保留以下 column
+  layer String? // values: "build" or ""
+
+  // 1-n
+  translations ClassroomTranslations[]
+  Schedule     Schedule[]
+}
+
+model ClassroomTranslations {
+  language Language
+
+  /// 教室詳細資料
+  detail String
+
+  classroomId String    @db.Uuid
+  classroom   Classroom @relation(fields: [classroomId], references: [id])
+
+  @@id([language, classroomId])
+}
+
+/// 實習分組，來自舊系統 NOL table `cougrp`
+model PracticeGroup {
+  id            String @id @db.Uuid
+  // parsed with `semester`+`identifier`
+  courseId      String @db.Uuid
+  ///組別
+  group         String @db.VarChar(1)
+  /// 學生人數
+  studentNumber Int
+  /// 教學助理人數
+  taNumber      Int
+
+  // TODO: 設計課程時間關聯
+  // schedule Schedule[]
+
+  translations PracticeGroupTranslations[]
+
+  course Course @relation(fields: [courseId], references: [id])
+}
+
+model PracticeGroupTranslations {
+  language        Language
+  practiceGroupId String   @db.Uuid
+
+  comment String
+
+  practiceGroup PracticeGroup @relation(fields: [practiceGroupId], references: [id])
+
+  @@id([language, practiceGroupId])
+}
+
+// ########################################
+//  核心能力相關
+// ########################################
+
+// TODO: fix relations, translation model
+
+// model CourseCapability {
+//   /// 課程識別碼
+//   courseId String @db.VarChar(9)
+//   // 待確認是否需要
+//   // deparmentId String @db.VarChar(4)
+//   /// 建議修課年級，[1-7]
+//   recommendedGrades Int[]  
+// }
+
+/// 系所核心能力
+model DeparmentCapability {
+  /// 系所代碼
+  deparmentId    String @db.VarChar(4)
+  /// 核心代碼
+  capabilityCode String @db.VarChar(1)
+
+  @@unique([deparmentId, capabilityCode])
+}
+
+/// 核心能力類型
+model CapabilityType {
+  /// 核心代碼， [A-Z]
+  code        String @id @db.VarChar(1)
+  /// 中文名稱
+  nameZhTw    String @db.VarChar(100)
+  /// 英文名曾
+  nameEn      String @db.VarChar(4000)
+  /// 中文說明
+  commentZhTw String @db.VarChar(150)
+  /// 英文說明
+  commentEn   String @db.VarChar(4000)
+}
+
+/// 課程分類
+model CourseCustomType {
+  /// 課程分類代碼， [A-Z]
+  code     String @id @db.VarChar(1)
+  /// 中文名稱
+  nameZhTw String @db.VarChar(50)
+  /// 英文名曾
+  nameEn   String @db.VarChar(50)
+}
+
+// ########################################
+//  領域專長
+// ########################################
+
+/// 領域專長說明主表
+model Domain {
```

Comment:
`Domain` can not be specified what you are meaning in 領域專長 in Chinese. I knew that 教務處 is using **Specialization Program** as the naming for 領域專長? Do you ever consider that as the name for this table. I know the `Specialization Program` doesn't make any sense at all, but we better to normalized the naming convention to avoid misunderstanding in the future...

@jc-hiroto What do you think?

Reply:
Following the convention sounds good to me.

Reply:
<del>But the convention is insane.</del>